packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -41,8 +41,10 @@ export const DatePickerMixin = (subclass) =>
          * Supported date formats:
          * - ISO 8601 `"YYYY-MM-DD"` (default)
          * - 6-digit extended ISO 8601 `"+YYYYYY-MM-DD"`, `"-YYYYYY-MM-DD"`
+         * - 24 or 27 extended ISO 8601 `YYYY-MM-DDTHH:mm:ss.sssZ` or `±YYYYYY-MM-DDTHH:mm:ss.sssZ`
+         * - A Date object
          *
-         * @type {string}
+         * @type {string | Date}
```

Comment:
This looks very similar to https://github.com/vaadin/vaadin-date-picker/pull/449

Also has the same issue about the value type changing on the fly:
![Screenshot 2022-05-30 at 9 27 13](https://user-images.githubusercontent.com/1222264/170930771-3d2b4f98-65b3-4455-ba27-21ffefd8a920.png)

If someone wants to use a `Date` to set the value, I guess they could use a helper library, such as date-fns, for converting the value first:

```html
<vaadin-date-picker .value="${format(this.date, 'yyyy-mm-dd')}`">
```

Reply:
Yes is similar, but those days we were not using type-safe, though, now that we have type definitions and union types, and the fact that we are encouraging to use TS in both flow frontend and hilla makes more sense.
Anyways, Let's revert the Date feature, and separate the discussion from this fix.

Reply:
Please change the PR title to reflect the new content (`fix: ...`)